### ISSUE TITLE
Time sync not writing expected time

### ISSUE THREAD

**Issue by cverges** [dnp3/opendnp3#35]
When connecting an Automatak DNP3 master to a GEC DNP3 outstation, the following is seen during the time sync 'write' packet:

![screen shot 2013-11-04 at 9 29 00 am](https://f.cloud.github.com/assets/743898/1466790/c572e30c-4576-11e3-9639-7e32a00f1b5f.png)

Note that the master's system clock is currently set to Mon Nov  4 09:28:26 PST 2013 of that screenshot.

**Comment by cverges** [comment]
http://stackoverflow.com/questions/16177295/get-time-since-epoch-in-milliseconds-preferably-using-c11-chrono

```
diff --git a/cpp/src/opendnp3/ITimeSource.cpp b/cpp/src/opendnp3/ITimeSource.cpp
index 59cb41d..34785dd 100644
--- a/cpp/src/opendnp3/ITimeSource.cpp
+++ b/cpp/src/opendnp3/ITimeSource.cpp
@@ -30,7 +30,7 @@ namespace opendnp3

 millis_t ITimeSource::GetMillisecondsSinceEpoch()
 {
-       return std::chrono::duration_cast<std::chrono::milliseconds>(this->GetUTC().time_since_epoch()).count();
+    return this->GetUTC().time_since_epoch() / std::chrono::milliseconds(1);
 }

 }
```

**Comment by cverges** [comment]
The system_clock doesn't appear to be used anywhere in the C++ code.  I'm suspecting that the initial time being used is based on some monotonic counter, like system uptime, rather than a wall clock, like system_clock.

**Comment by cverges** [comment]
Resolved in the 1.1.x branch.
